### DISCUSSION TITLE
add a -remove subargument for alert command

### DISCUSSION THREAD

**Discussion by CloudeLecaw** [Zrips/CMI#7859] [Enhancement]
currently we got the command
```
cmi alert (playerName) (Reason)
```

we are using an external plugin to do timed things and execute commands when the time is over

so to not nees to remove alerts each time from the chat interface it would be nice so have a remove arguments that removes the alert by command.

so i suggest
```
cmi alert (playerName) (Reason) (-remove) (-s)
```

permission ```cmi.commamd.alert.remove```

when the -remove Argument is used the alert (or alert of the player if defined) should be removed.

and about the -s

well i think i dont need to explain what it should be able to do. so many command allready have it. so it would be a nice extra for this command =)

Before replying "But we allready can remove it with using the command aggain".. thats not what i ask for and not the thing i would need here.. and is not content of this request. i asked for -remove as subargument for removal.

**Comment by mrfloris** [comment]
Hey there,

Can you please explain why you want a separate alert parameter specifically? How does running the command again not achieve what you're trying to do?

<img width="970" alt="Screen-Shot 2023-05-27 at 13 25 13" src="https://github.com/Zrips/CMI/assets/28841349/d9253647-2aa6-4fcc-9769-5688a8bed678">

You can see in the screenshot you can run a console command to add (when not yet on the list) and remove (when already on the list) someone. This should be sufficient for someone to remove someone before the 24 hours are up. 

I am perfectly fine if it wasn't a thing yet to agree that it would be nice to have the option to remove it via a command. But we can already, so I am curious to hear why it's not helping you achieve what you need out of this. 

I can only think of "forced" removal, even if they aren't on the list. But does that mean we should also have a forced -add just in case we are in the opposite situation? 

Looking forward to understanding why you can't run the same command again to remove it in a scheduler. Not here to 'argue' your request. Merely trying to understand why a sub-command might be desirable.

**Comment by mrfloris** [comment]
I think I would _personally_ not object against revamping the approach to the alert command as a hole. 

Throw away the old alertlist command, make it the default for alert, and have the alert with just a username report back if someone's on the list, and if so what expiry time (and option to remove), or report back they're not on the list with an option to add. And remove the old functionality of the toggle to auto add,remove, and instead require params. So when we type -add 3x, it will only add them once, and at most re-add them, and not auto remove them. 

So we will get:
```
- remove existing command: /cmi alertlist, and move functionality/behavior to /cmi alert
```
```
/cmi alert 
(shows list of all added names, with option to bump (add) to set the expiry time to a day again) or [x] to remove their listing. 
Output:
--------------------------------------------
[x] [bump] 01 mrfloris (12 hours and 10 minutes) (reason goes here)
[x] [bump] 02 zrips (13 hours and 10 minutes) (reason goes here)
[x] [bump] 03 roberto (42 hours and 10 minutes) (reason goes here)
--------------------------------------------
(clicking reason could ask in the chat "Type a new reason:")
```
```
/cmi alert [playerName]
(shows if the user is on the list, if so, with option to bump (add) to set the expiry time to a day again) or [x] to remove their listing. - and show how much expiry time is left)
Output:
--------------------------------------------
[x] [bump] 02 zrips (13 hours and 10 minutes) (reason goes here)
--------------------------------------------
(clicking alert could ask in chat to set a reason)

(shows if they are Not on the list, with option to [add] to the list)
Output:
--------------------------------------------
Alerts for zrips not set, -> [add] 
--------------------------------------------
```
```
/cmi alert [playerName] -add (reason)
(forcefully adds the person to the list, same as 'bumping' their time to 24 hours again, if they are already on the list, do not remove them)
Output:
Alerts for zrips set (expires in a day) [remove]
```
```
/cmi alert [playerName] -remove
(forcefully remove someone from the list. If they aren't on the list, just do nothing, at most report back they weren't on the list. Do not add them to the list.)
Output:
Alerts for zrips removed [add]
```
```
And a new optional new parameter if it doesn't exist yet; 
/cmi alert <params> -s
(to silence the output so we can create our own custom alias commands with our own output, and use it in scheduler.yml and eventcommands.yml etc)
```

<img width="973" alt="Screen-Shot 2023-05-27 at 13 43 49" src="https://github.com/Zrips/CMI/assets/28841349/0aa966eb-76cc-4902-a073-9c2bcee0bd43">

It's a bit messy now.

**Comment by CloudeLecaw** [comment]
aii you got brillant ideas <3

yes i know it can be removed with the current `toggle` behavior but for the external useage i feeled like to have a clear forced removal.
i want to clearly trigger a remove, even no alaert exist (because it might was removed allready by other staffteam for example. with the current alart System this would create an alert because none exist before. worst thing with a forced removal Argument that may could happen is that you get a message `no alerts set to this player`

well yiu said your examples looks messy? 
i dont think so.. well it looks like you totally understood and showed with nice example what i wanted to suggest =)

**Comment by mrfloris** [comment]
CMI version 9.5.1.3 has some of our suggestions now implemented. 
https://www.spigotmc.org/resources/cmi-298-commands-insane-kits-portals-essentials-economy-mysql-sqlite-much-more.3742/update?update=502220
